conda_vendor/conda_vendor.py

```diff
@@ -227,36 +216,47 @@ def solve_environment(
     platform: str
         platform to vendor for: eg. linux-32, win-64, osx-64, etc.
 
+    virtual_package_spec: pathlib.Path or str
+        location of a conda-lock virtual package specification yaml.  See
+        the documentation for conda-lock for the correct format.  If
+        this is None, use the default virtual packages that conda-lock
+        provides for {platform}
+
     Returns
     -------
     list [ conda_lock.conda_solver.FetchAction ]
 
     """
     assert isinstance(environment_file, Path)
 
-    # generate conda-lock's LockSpecification
-    lock_spec = _parse_environment_file(environment_file, platform)
+    # generate conda-lock's LockSpecification, this will parse the environment
+    # file for us and give us more easily handled requirements.
+    lock_spec = _generate_lock_spec(environment_file, platform)
     specs = _get_query_list(lock_spec)
 
+    # find virtual packages or use defaults
+    virt_pkgs = _get_virtual_packages(virtual_package_spec)
+
     _cyan(f"Using Solver: {solver}", bold=False)
     _cyan(f"Solving for Platform: {platform}", bold=False)
     _cyan(f"Solving for Spec: {specs}", bold=False)
+    _cyan("Virtual Packages:", bold=False)
+    for _, pkg in virt_pkgs.all_repodata[platform]["packages"].items():
```

Comment:
you could use .values() instead of .items() since the keys aren't used

Reply:
nice.  this is why we code review.  i don't think i ever knew about `.values()`

Reply:
changed.